Digraphy/Digraphy/Controllers/TodoController.cs

```diff
@@ -0,0 +1,127 @@
+using AutoMapper;
+using Digraphy.Dto;
+using Digraphy.Models;
+using Digraphy.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Digraphy.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class TodoController : Controller
+{
+    private readonly ITodoRepository _todoRepository;
+    private readonly IMapper _dtoMapper;
+
+    public TodoController(ITodoRepository todoRepository, IMapper mapper)
+    {
+        _todoRepository = todoRepository;
+        _dtoMapper = mapper;
+    }
+    
+    [HttpGet]
+    [ProducesResponseType(200, Type = typeof(IEnumerable<Todo>))]
+    public IActionResult GetTodos()
+    {
+        var todos = _dtoMapper.Map<List<TodoDto>>(_todoRepository.GetTodos());
+
+        if (!ModelState.IsValid)
+            return BadRequest(ModelState);
+
+        return Ok(todos);
+    }
+    
+    [HttpGet("{todoId}")]
+    [ProducesResponseType(200, Type = typeof(Todo))]
+    [ProducesResponseType(400)]
+    public IActionResult GetTodo(int todoId)
+    {
+        if (!_todoRepository.TodoExists(todoId))
+            return NotFound();
+        
+        var todo = _dtoMapper.Map<TodoDto>(_todoRepository.GetTodo(todoId));
+    
+        if (!ModelState.IsValid)
+            return BadRequest(ModelState);
+    
+        return Ok(todo);
+    }
+    
+    [HttpPut("{todoId}")]
+    [ProducesResponseType(400)]
+    [ProducesResponseType(204)]
+    [ProducesResponseType(404)]
+    public IActionResult UpdateTodo(int todoId,
+        [FromBody] PutTodoDto updatedTodo)
+    {
+        if (updatedTodo == null)
+            return BadRequest(ModelState);
```

Comment:
I am unable to test it properly atm since there's no database, the ide is telling me to get rid of it, so I'm thinking we just add it back in later if it becomes a problem